Stresser/ArtifactManager.cpp

```diff
@@ -0,0 +1,45 @@
+#include "ArtifactManager.h"
+
+ArtifactManager::ArtifactManager(const std::string& endpointId, const HANDLE& shutdownEvent, const EndpointController& endpointController,
+		const PolicyController& policyController, const RuleController& ruleController) :
+endpointId(endpointId), m_changePolicyEvent(CHANGE_POLICY), m_shutdownEvent(shutdownEvent)
+{
+	this->changePolicyThread = StandardThread([this, &endpointController, &policyController, &ruleController]()
+	{
+		this->changePolicyThreadFunction(endpointController, policyController, ruleController);
+	});
+}
+
+void ArtifactManager::changePolicyThreadFunction(const EndpointController& endpointController,
```

Comment:
This implementation prevents the Single-responsibility Principle.

- `ArtifactManager` responsible to manage the application fake artifacts.
- Fetch policy and update the rules is another responsibility.

We should move this thread function outside of the scope of this class and "register" the artifacts manager to "receive" notification when "policy-change" happened.

Reply:
Optional solution:
- Create class `PolicyNotifications` (or other cool name)
- This class act as "Publisher" for policy-notifications 
- Create `Subscriber` interface which expose function to run when notification happened
- Allow others (`Subscribers`) to register for notifications (the class manage list of subscribers)
- Implement `changePolicyThreadFunction` inside the "PolicyNotification" class 
- Implement `Subscriber` interface in `ArtifactManager` to what ever you need when notification happened